code/wdb/datastripe.q

```diff
@@ -45,30 +76,29 @@ upserttopartition:{[dir;tablename;keycol;enumdata;nextp]
 	  (directory;enumdata);
 	  {[e] .lg.e[`upserttopartition;"Failed to save table to disk : ",e];'e}
 	];
-	};	
+	};
 
 savetablesoverperiod:{[dir;tablename;nextp]
-	/- function to get keycol for table from access table
-	keycol:`sym; / variable keycol functionality added in TCS-73
-	/- get distinct values to partition table on
-	partitionlist:raze value each ?[[`.]tablename;();1b;enlist[keycol]!enlist keycol];
-	/- enumerate table to be upserted and get each table by sym
-	enumdata:{[dir;tablename;keycol;nextp;s] .Q.en[dir;0!?[[`.]tablename;((<;`time;nextp);(=;keycol;enlist s));0b;()]]}[dir;tablename;keycol;nextp]'[partitionlist];
-	/-upsert table to partition
-	upserttopartition[dir;tablename;keycol;;nextp] each enumdata;
-	/- delete data from last period
-	.[.ds.deletetablebefore;(tablename;`time;nextp)];
-	/ .[{![x;enlist(<;`time;y);0b;0#`]};(tablename;nextp)];
-	/- run a garbage collection (if enabled)
-	.gc.run[];
-	};
+    /- function to get keycol for table from access table
+    keycol:$[.wdb.tablekeycols[tablename]=`;`sym;.wdb.tablekeycols[tablename]];
+    /- get distint values to partition table on
+    partitionlist:raze value each ?[[`.]tablename;();1b;enlist[keycol]!enlist keycol];
+    /- enumerate table to be upserted and get each table by sym
+    enumdata:{[dir;tablename;keycol;nextp;s] .Q.en[dir;0!?[[`.]tablename;((<;`time;nextp);(=;keycol;enlist s));0b;()]]}[dir;tablename;keycol;nextp]'[partitionlist];
+    /-upsert table to partition
+    upserttopartition[dir;tablename;keycol;;nextp] each enumdata;
+    /- delete data from last period
+    .[.ds.deletetablebefore;(tablename;`time;nextp)];
+    .[{![x;enlist(<;`time;y);0b;0#`]};(tablename;nextp)];
```

Comment:
another instance were we could be using the defined version of `deletetablebefore` I think

actually that's what's in the line above - is this line a duplicate?

Reply:
yes, this is a duplicate. it has been removed now